router/http_test.go

```diff
@@ -339,6 +339,34 @@ func (s *S) TestHTTPHeadersFromClient(c *C) {
 	c.Assert(res.StatusCode, Equals, 200)
 }
 
+func (s *S) TestConnectionCloseHeaderFromClient(c *C) {
+	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Connection: close header should be stripped by the reverse proxy so it
+		// always does keep-alive with backends.
+		c.Assert(req.Close, Equals, false)
+		c.Assert(req.Header.Get("Proxy-Authenticate"), Equals, "")
+		c.Assert(req.Header.Get("Proxy-Authorization"), Equals, "")
```

Comment:
I'm not convinced that stripping these headers is correct. Per [RFC 7235](https://tools.ietf.org/html/rfc7235#section-4.4):

> the Proxy-Authorization header field applies only to the next inbound proxy that demanded authentication using the Proxy-Authenticate field.  When multiple proxies are used in a chain, the Proxy-Authorization header field is consumed by the first inbound proxy that was expecting to receive credentials.

Since we are not expecting credentials, do we need to "consume" it?